meeting-notes/2018/2018-09-10--js-core-dev-team-weekly.md

```diff
@@ -0,0 +1,144 @@
+
+## Notes
+
+### Week Update
+
+@alanshaw
+- Done:
+    - Published release candidate for `js-ipfs` 0.32.0-rc.1 and 0.32.0-rc.2
+    - Test stabilisation work [#1541](https://github.com/ipfs/js-ipfs/pull/1541)
+    - Sent PR for faster startup time when using CLI with no daemon running [#1542](https://github.com/ipfs/js-ipfs/pull/1542)
+    - Many fixes to API docs [#1538](https://github.com/ipfs/js-ipfs/pull/1538)
+    - Made progress adding `--cid-base` option to _all_ relevant `js-ipfs` commands (working in branch [feat/cid-base-option](https://github.com/ipfs/js-ipfs/tree/feat/cid-base-option))
+- Blocked:
+    - Jenkins npm errors frequent [#51](https://github.com/ipfs/testing/issues/51)
+    - Jenkins windows workers out of space [#135](https://github.com/ipfs/testing/issues/135)
+- Next:
+    - Release js-ipfs 0.32
+    - Finish `--cid-base` option
+    - Create interop tests for block store get by CIDv0/1
+    - Implement block store get by CIDv0/1
+
+@achingbrain
+- Done:
+  - Spun up npm registry at https://registry.js.ipfs.io
+    - Play along at home with `npm i --registry=https://registry.js.ipfs.io`
+  - Fixed flaky pubsub tests https://github.com/ipfs/interop/pull/33
+  - Made pin tests less flaky https://github.com/ipfs/interop/pull/34
+  - Consolidated HTTP API https://github.com/ipfs-shipyard/js-ipfs-http
+- Blocked:
+  - Create S3 backend for npm-on-ipfs: https://github.com/ipfs/infrastructure/issues/432
+  - Remove password requirement from libp2p-keychain? https://github.com/libp2p/js-libp2p-keychain/pull/25
+  - Support OTP in Aegir: https://github.com/ipfs/aegir/pull/260
+- Next:
+  - Continue HTTP API https://github.com/ipfs-shipyard/js-ipfs-http
+
+@jacobheun
+ - Done:
+   - Added core state machine logic for libp2p-switch. Started working on the connection state machine logic. Reviewed several existing issues surrounding the [state machine issue for libp2p](https://github.com/libp2p/js-libp2p/issues/98) to make sure that is incorporated.
+   - Have a basic demo of delegated routing working against a local instance of the supporting go-ipfs branch. Will demo today at IPFS All Hands.
+     - I believe there is a performance issue here when the numbner of providers is low. The http query stays open until at least 20 providers are found. I've seen this stay open for 30+ minutes.
+   - Patch release for libp2p-switch (0.40.8) for dependency updates, including multistream-select already piped fix version bump.   
+ - Blocked:
+   - Delegated routing support in [go-ipfs#4594](https://github.com/ipfs/go-ipfs/pull/4595).
+ - Next:
+   - libp2p-tcp transport isnt currently dialble over dns, looking into this (mafmt needs to be updated for this to work)
+   - Continue with state machine updates for libp2p connections
+
+@vasco-santos
+- Done:
+  - IPNS over Pubsub OKR
+    - Created `js-datastore-pubsub` module
+      - Initial implementation of [ipfs/js-datastore-pubsub#1](https://github.com/ipfs/js-datastore-pubsub/pull/1)
```

Comment:
@vasco-santos, This design is incorrect. We don't want a datastore to publish every key that gets stored. 

Do you have a diagram of how things get plugged together? The IPNS system (module) should use PubSub to publish and subscribe to the new keys and then make the decision of storing them in the DataStore.

@vasco-santos do you have a diagram of your architecture? We can workshop this together if you need.